rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,221 @@
+import akka.http.scaladsl.model._
+import akka.pattern.pipe
+import akka.pki.kubernetes.PemManagersProvider
+import akka.rollingupdate.OlderCostsMore
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.GiveUp
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.PodAnnotated
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.RequestResult
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.RetryAnnotate
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.RetryTimerId
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.ScheduleRetry
+import akka.rollingupdate.kubernetes.PodDeletionCostAnnotator.toResult
+import akka.stream.Materializer
+import com.typesafe.config.Config
+
+import java.security.KeyStore
+import java.security.SecureRandom
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.KeyManager
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import scala.collection.immutable
+import scala.collection.immutable.SortedSet
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationLong
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Actor responsible to annotate the hosting pod with the pod-deletion-cost.
+ * It will automatically retry upon a fixed-configurable delay if the annotation fails.
+ */
+@InternalApi private[kubernetes] final class PodDeletionCostAnnotator(
+    settings: KubernetesSettings,
+    apiToken: String,
+    podNamespace: String,
+    costSettings: PodDeletionCostSettings)
+    extends Actor
+    with ActorLogging
+    with Timers {
+  private val cluster = Cluster(context.system)
+  private val http = Http()(context.system)
+
+  Cluster(context.system).subscribe(context.self, classOf[ClusterEvent.MemberUp], classOf[ClusterEvent.MemberRemoved])
+
+  private lazy val sslContext = {
+    val certificates = PemManagersProvider.loadCertificates(settings.apiCaPath)
+    val factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+    val keyStore = KeyStore.getInstance("PKCS12")
+    keyStore.load(null)
+    factory.init(keyStore, Array.empty)
+    val km: Array[KeyManager] = factory.getKeyManagers
+    val tm: Array[TrustManager] =
+      PemManagersProvider.buildTrustManagers(certificates)
+    val random: SecureRandom = new SecureRandom
+    val sslContext = SSLContext.getInstance("TLSv1.2")
+    sslContext.init(km, tm, random)
+    sslContext
+  }
+  private val clientSslContext: Option[HttpsConnectionContext] = if (settings.secure) Some(ConnectionContext.httpsClient(sslContext)) else None
+
+  implicit val dispatcher: ExecutionContextExecutor = context.system.dispatcher
+  def receive = onMessage(0, SortedSet.empty(Member.ageOrdering))
+
+  private def onMessage(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
```

Comment:
Maybe give this a more clear name to hint about the state machine, I think we often use `idle` for such cases.